src/control/server/harness_test.go

```diff
@@ -235,23 +240,132 @@ func TestServer_HarnessGetMSLeaderInstance(t *testing.T) {
 					t.Fatal(err)
 				}
 			}
-			h.setStarted()
+			h.started.SetTrue()
 
 			_, err := h.GetMSLeaderInstance()
-			common.CmpErr(t, tc.expError, err)
+			CmpErr(t, tc.expError, err)
 		})
 	}
 }
 
 func TestServer_HarnessIOServerStart(t *testing.T) {
+	defaultAddrStr := "127.0.0.1:10001"
+	defaultAddr, err := net.ResolveTCPAddr("tcp", defaultAddrStr)
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	for name, tc := range map[string]struct {
-		trc           *ioserver.TestRunnerConfig
-		expStartErr   error
-		expStartCount int
+		trc              *ioserver.TestRunnerConfig
+		isAP             bool           // should first instance be AP/MS replica/bootstrap
+		rankInSuperblock bool           // rank already set in superblock when starting
+		instanceUuids    map[int]string // UUIDs for each instance.Index()
+		expStartErr      error
+		expStartCount    int
+		expDrpcCalls     map[uint32][]int32     // method ids called for each instance.Index()
+		expGrpcCalls     map[uint32][]string    // string repr of call for each instance.Index()
+		expRanks         map[uint32]system.Rank // ranks to have been set during Start()
+		expMembers       system.Members         // members to have been registered during Stop()
+		expIoErrs        map[uint32]error       // errors expected from instances
 	}{
 		"normal startup/shutdown": {
```

Comment:
Looking a little more closely at these tests, still a little hard to follow. I guess it's pretty weird to me that some cases are normal startup/shutdown but still return errors. Would be nice to add some commentary about why we're triggering errors in a normal case - Are we just using the context timeout to end the test case?

Reply:
yes, because they wouldn't normally end in normal circumstances

Reply:
Ah, got it. Thanks for the details, Tom.